fennel.lua

```diff
@@ -1221,10 +1245,23 @@ SPECIALS['fn'] = function(ast, scope, parent)
             assertCompile(false, 'expected symbol for function parameter', ast)
         end
     end
+    if type(ast[index + 1]) == 'string' and index + 1 < #ast then
+      index = index + 1
+      docstring = ast[index]
+      if rootOptions.useMetadata then
+        docstring = docstring -- account for newlines and nested [[ ]] multiline strings
+                :gsub('\n', '\\n')
+                :gsub('[[](=*)[[]', '[=%1['):gsub('[]](=*)[]]', ']=%1]')
+        -- splice in argslist and format docstring contents
+        local argSep = #argNameList > 0 and ' ' or ''
+        local invokeStr = string.format("(%s%s%s)", fnName, argSep, table.concat(argNameList, ' '))
```

Comment:
What if the docstrings and the arg names were stored in different metadata slots? That way when you did doc lookup on a function, you could get argument names even on functions that didn't have docstrings.

Reply:
Good call - I was a bit undecided about doing that right out of the gate but it absolutely makes more sense to do it that way. Plus, having those other items in the metadata will probably open up other use-cases.